lib/Atat/Objects.pm

```diff
@@ -0,0 +1,96 @@
+# Domain model.
+#
+# This file describes our domain using an object model.
+#
+# The following terminology may be helpful:
+#     * 'package' defines a class.
+#     * A class 'has' an attribute, which may be another object (delegation).
+#     * Plural attributes imply a 1-many (or many-many) relationship.
+#     * A class 'is' another class (vertical composition).
+#     * A class 'does' a role (horizontal composition).
+
+# User: An ATAT user.
+package User {
+  use Mojo::Base -base;
+  has 'username';    # Username in ATAT
+  has 'piv';         # Data from PIV card.
+  has 'roles';       # Roles within ATAT.
+}
+
+# UserRole: Defines a set of permissions surrounding access to a CSP.
+package UserRole {
+  use Mojo::Base -base;
+  has 'csp_account';    # CSP account (optional)
+  has 'permissions';    # List of permissions (strings) for this role.
+}
+
+# Account: An account.
+package Account {
+  use Mojo::Base -base;
+  has 'csp_role';       # The role in the CSP.
+  has 'token';          # Auth token.
+}
+
+# Grant: A permission that a person has for a bucket.
+package Grant {
+  use Mojo::Base -base;
+  has 'user';           # An ATAT user.
+  has 'role';           # A UserRole (above).
+}
```

Comment:
I would assume that a Grant needs a 'status' - such as Pending, Accepted, Rejected   and a 'history log' - Accepted on Jan 21st by user J.

Reply:
Uuuh... I think we're getting ahead of ourselves on this one. We're going to have to talk a _lot_ more about approvals, acceptance, rejection, etc. This is just defining the relationship.

Reply:
(It's also a throw away model so we can get our feet wet, right? In which case, I don't care about logs, logs will come in time.)